crates/astria-core/src/sequencer/v1alpha1/block.rs

```diff
@@ -366,6 +393,25 @@ impl SequencerBlock {
         }
     }
 
+    #[must_use]
+    pub fn filtered_block(self, rollup_ids: Vec<RollupId>) -> Self {
+        let mut rollup_transactions = IndexMap::with_capacity(rollup_ids.len());
+        for id in rollup_ids {
+            let Some(rollup_transaction) = self.rollup_transactions.get(&id) else {
```

Comment:
naming here is a bit weird since this is still multiple txs right? maybe name `rollup_data` or something?